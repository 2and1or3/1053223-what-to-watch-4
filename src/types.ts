interface FilmType {
  id: number;
  title: string;
  preview: string;
  background: string;
  backgroundColor: string;
  cover: string;
  genre: string;
  isFavorite: boolean;
  src: string;
  release: string;
  description: string;
  poster: string;
  duration: number;
  rating: number;
  voiceCount: number;
  director: string;
  actors: string[];
}

interface CommentType {
  id: string;
  author: string;
  date: string;
  description: string;
  rate: string;
}

interface GenreType {
  id: string;
  title: string;
}

type RefType = () => void | {current: React.ReactNode};

export {FilmType, CommentType, RefType, GenreType};
