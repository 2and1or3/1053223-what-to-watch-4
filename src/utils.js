const createVideoMock = (element) => {
  if (element.type === `video`) {
    return element;
  }

  return null;
};

export {createVideoMock};
