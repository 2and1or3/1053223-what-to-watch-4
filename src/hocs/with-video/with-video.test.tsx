import * as React from "react";
import * as renderer from "react-test-renderer";

import {createVideoMock} from '../../utils';
import withVideo from './with-video';
import {FilmType} from '../../types';

const film: FilmType = {
  id: `0`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000`,
  cover: `img/the-grand-budapest-hotel-poster.jpg`,
  isFavorite: false,
  src: `path`,
  genre: `Drama`,
  release: `2014`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  voiceCount: 240,
  duration: 99,
  director: `Wes Andreson`,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`],
};

interface MockProps {
  children: React.ReactNode;
}

const MockComponent: React.FunctionComponent<MockProps> = (props: MockProps) => {
  const {children} = props;
  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWithVideo = withVideo(MockComponent);

it(`Render Component with video`, () => {
  const tree = renderer
  .create(
      <MockComponentWithVideo
        currentFilm = {film}
        isMuted = {true}
        isPlaying = {false}
      />, {
        createNodeMock: createVideoMock
      })
  .toJSON();

  expect(tree).toMatchSnapshot();
});
