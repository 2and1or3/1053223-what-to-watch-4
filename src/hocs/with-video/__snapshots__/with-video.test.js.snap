// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component with video 1`] = `
<div>
  <video
    height="175"
    width="280"
  />
</div>
`;
